### PR TITLE
Construct `FpVar` from bits

### PR DESCRIPTION
**Pull request by Pratyush**
This is useful for Poseidon-like sponges.

### CODE REVIEW

File: src/bits/boolean.rs [hunk 1]
```diff
@@ -608,6 +608,56 @@ impl<F: Field> Boolean<F> {
...
+                    } else {
+                        // On the first iteration, set the value to be the first bit.
+                        value = Some(F::from(b));
+                    }
+                    power.double_in_place();
+                }
+            }
+        }
+
+        if bits.is_constant() {
```
Comment (weikengchen) [inline src/bits/boolean.rs:638]: One question. If CS is in the setup mode, `value` would be `None`. Would it panic here due to `unwrap`?

Comment (Pratyush) [reply src/bits/boolean.rs:638]: Ah that's a good point. I'll fix it now

File: src/bits/boolean.rs [hunk 2]
```diff
@@ -608,6 +608,56 @@ impl<F: Field> Boolean<F> {
...
+    /// Convert a little-endian bitwise representation of a field element to `FpVar<F>`
+    #[tracing::instrument(target = "r1cs", skip(bits))]
+    pub fn le_bits_to_fp_var(bits: &[Self]) -> Result<FpVar<F>, SynthesisError>
+    where
+        F: PrimeField,
+    {
+        // Compute the value of the `FpVar` variable via double-and-add.
+        let mut value = None;
+        let cs = bits.cs();
+        if !cs.is_in_setup_mode() {
```
Comment (ValarDragon) [inline src/bits/boolean.rs:620]: is there a `[Bool] -> Vec<u8>` method? If so then we could just do that + `F::from_bytes_mod_order` to get the value. (its more efficient field airthmetic wise than what is here)

File: src/bits/boolean.rs (hunk 1 above)
Comment (Pratyush) [reply src/bits/boolean.rs:639]: Fixed.

File: src/bits/boolean.rs (hunk 2 above)
Comment (Pratyush) [reply src/bits/boolean.rs]: Done!

**Comment by weikengchen** [comment]
Thanks! (Sorry that I am slower in response due to another project :))

**Comment by Pratyush** [comment]
No worries!

**Pull request merged**